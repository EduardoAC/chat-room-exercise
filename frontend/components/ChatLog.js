import React from 'react';
import PropTypes from 'prop-types';

import '../styles/ChatLog.css';

const ChatLog = ({ messages }) => {
  const messageLog = (messages) ? messages.map((msg) => (
    <li
      key={msg}
      className="chat-message-log__message"
    >
      {msg}
    </li>
  )) : null;
  return (<ul className="chat-message-log">{messageLog}</ul>);
};

ChatLog.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.string),
}

export default ChatLog;
