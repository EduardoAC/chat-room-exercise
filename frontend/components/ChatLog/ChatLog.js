import React from 'react';
import PropTypes from 'prop-types';

const ChatLog = ({ messages }) => {
  const messageLog = (messages) ? messages.map((msg) => (
    <li key={msg}>{msg}</li>
  )) : null;
  return (<ul>{messageLog}</ul>);
};

ChatLog.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.string),
}

export default ChatLog;
