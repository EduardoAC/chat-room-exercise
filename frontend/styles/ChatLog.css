.chat-message-log {
  border: 2px solid #bfbfbf;
  border-radius: 4px 4px 0 0;
  height: 200px;
  list-style-type: none;
  margin: 0;
  overflow: auto;
  padding: 7px;
  width: 100%;
  word-break: break-all;
}

.chat-message-log__message::before {
  content: '> ';
}
