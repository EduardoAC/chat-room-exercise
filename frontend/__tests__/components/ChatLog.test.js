import React from 'react';
import { shallow } from 'enzyme';

import ChatLog from '../../components/ChatLog';

describe('<ChatLog />', () => {
  it('Component renders correctly', () => {
    const chatLogWrapper = shallow(<ChatLog messages={[]} />);
    expect(chatLogWrapper).toMatchSnapshot();
  });

  it('should render the message', () => {
    const messages = ['hello'];
    const chatLogWrapper = shallow(<ChatLog messages={messages} />);
    const messageList = chatLogWrapper.find('li');
    expect(messageList).toHaveLength(1);
    // Since we verified that the content is one, we can access to the element
    // as dom element without extra functions like fitst() or at(0)
    expect(messageList.text()).toBe('hello');
  });

  it('should render all the messages passed in the right order', () => {
    const messages = ['hello', 'world', '!'];
    const chatLogWrapper = shallow(<ChatLog messages={messages} />);
    const messageList = chatLogWrapper.find('li');
    expect(messageList).toHaveLength(3);
    messages.forEach((value, index) => {
      expect(messageList.at(index).text()).toBe(value);
    });
  });

});
