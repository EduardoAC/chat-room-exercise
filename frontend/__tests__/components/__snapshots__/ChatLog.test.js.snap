// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatLog /> Component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="chat-message-log"
>
    
</ul>,
  "nodes": Array [
    <ul
      className="chat-message-log"
>
      
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ChatLog
        messages={Array []}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "messages": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="chat-message-log"
>
          
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="chat-message-log"
>
          
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ChatLog
    messages={Array []}
/>,
}
`;
